torch/_library/custom_ops.py

```diff
@@ -594,19 +594,13 @@ def fake_impl(*args, **kwargs):
 
         schema = self._opoverload._schema
         if schema.is_mutable:
+            mutated_idxs, mutated_keys = utils.mutated_args_kwargs(schema)
 
             def adinplaceorview_impl(keyset, *args, **kwargs):
-                for arg, val in utils.zip_schema(schema, args, kwargs):
-                    if not arg.alias_info:
-                        continue
-                    if not arg.alias_info.is_write:
-                        continue
-                    if isinstance(val, Tensor):
-                        torch.autograd.graph.increment_version(val)
-                    elif isinstance(val, (tuple, list)):
-                        for v in val:
-                            if isinstance(v, Tensor):
-                                torch.autograd.graph.increment_version(v)
+                for idx in mutated_idxs:
+                    increment_version(args[idx])
+                for key in mutated_keys:
+                    increment_version(kwargs[key])
```

Comment:
if we're worried about python overhead here, `increment_version()` should (as of recently) support `Iterable[Tensor]` as an argument: https://github.com/pytorch/pytorch/blob/main/torch/autograd/graph.py#L226

Reply:
nice, thanks for pointing that out

Reply:
It's not clear to me if building an iterator to pass to increment_version is less expensive than calling increment_version in a loop, but I'll try it out